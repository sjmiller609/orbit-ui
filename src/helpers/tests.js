import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
