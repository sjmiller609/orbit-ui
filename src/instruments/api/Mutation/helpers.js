const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
