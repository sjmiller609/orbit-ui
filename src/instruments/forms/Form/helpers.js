// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
