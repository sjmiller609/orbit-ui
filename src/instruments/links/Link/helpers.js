// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
