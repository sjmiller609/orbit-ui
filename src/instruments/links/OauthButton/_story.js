import React from 'react'
import { storiesOf } from '@storybook/react'
import OAuthButton from './'

storiesOf('Instruments|Links', module).add('OAuthButton', () => (
  <OAuthButton>Mock link</OAuthButton>
))
