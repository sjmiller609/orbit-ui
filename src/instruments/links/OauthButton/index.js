'use strict'
import PropTypes from 'prop-types'
import React from 'react'
import { Link, Row, Box, Icon, LoadImg } from 'instruments'
import s from './styles.scss'
import classnames from 'classnames'

const services = {
  google: {
    text: ' with Google',
    img: 'google.svg',
    className: 'google',
  },
  github: {
    text: ' with Github',
    img: 'github.svg',
    className: 'github',
  },
  auth0: {
    text: ' with Auth0',
    img: 'auth0.svg',
    className: 'auth0',
  },
}

const OauthButton = ({ service, login, to, className, ...props }) => {
  const oauth = services[service]
  const Img = LoadImg(() => import(`./img/${oauth.img}`))
  return (
    <Link
      {...props}
      to={to}
      title={oauth.text}
      newTab={false}
      className={classnames(s.button, s[oauth.className], className)}>
      <Row full justify="flex-start">
        <Box className={s.img}>
          <Img />
        </Box>
        <span className={s.text}>
          {login ? 'Login' : 'Sign up'}
          {oauth.text} <Icon className={s.arrow} icon="arrow_darkBg" />
        </span>
      </Row>
    </Link>
  )
}

OauthButton.propTypes = {
  service: PropTypes.string,
  login: PropTypes.bool,
  to: PropTypes.string,
  className: PropTypes.string,
}

export default OauthButton
