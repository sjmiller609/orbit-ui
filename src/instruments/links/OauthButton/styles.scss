@import 'src/styles/mixins.scss';

.button {
  display: inline-block;
  @include filmo;
  white-space: nowrap;
  border-radius: 3px;
  width: 100%;
  // padding: 1.4em 1.8em 1.05em;
  height: 4em; //specifying for 100% children

  @include shadow;
  overflow: hidden;
  background-image: linear-gradient(to right, $green, $blue);
  &,
  &:visited {
    color: $white;
  }
  .text {
    height: 100%;
    padding: 1.4em 1.8em 1.05em;
    border-left: 2px solid $white;
  }

  &.google {
    background-image: linear-gradient(to right, $darkBlue, $lightBlue);
  }
  &.github {
    background-image: linear-gradient(to right, $darkPurple, $lightPurple);
  }
  &.auth0 {
    background-image: linear-gradient(to right, $red, $darkOrange);
  }
}

.img {
  height: 100%;
  width: 4em;

  img {
    height: 52%;
  }
}

.arrow {
  display: inline-block;
  height: 0.75em;
  margin-left: 1rem;

  @media screen and (max-width: 380px) {
    display: none;
  }
}
