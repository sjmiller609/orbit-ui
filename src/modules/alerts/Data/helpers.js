import { formErrors } from 'instruments'

const errors = [
  {
    key: 'permissions',
    name: 'properties.alert_emails',
    error: 'Contact your workspace admin to upgrade your permissions.',
  },
]

export const handleError = error => formErrors({ error, errors })

export const trimError = error => error.split(':')[1].trim()
