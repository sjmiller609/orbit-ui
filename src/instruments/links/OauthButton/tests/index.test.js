import React from 'react'
import { mountWrap } from 'helpers/tests'
import OauthButton from '../index'

const props = {
  service: 'google',
  displayName: 'Google',
  login: false,
  to: '/',
  className: 'testClassString',
}

describe('OauthButton', () => {
  it('should be defined', () => {
    expect(OauthButton).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = mountWrap(<OauthButton {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render login correctly', () => {
    const wrapper = mountWrap(<OauthButton {...props} login={true} />)
    expect(wrapper).toMatchSnapshot()
  })
})
