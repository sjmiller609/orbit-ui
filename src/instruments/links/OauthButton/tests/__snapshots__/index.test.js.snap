// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OauthButton should render correctly 1`] = `
<OauthButton
  className="testClassString"
  displayName="Google"
  login={false}
  service="google"
  to="/"
>
  <Row
    className="testClassString"
    full={true}
  >
    <Flex
      className="testClassString"
      flow="row"
      full={true}
    >
      <div
        className="flex row undefined full testClassString"
      >
        <Link
          className="button google"
          newTab={false}
          target="_self"
          title="Sign up with Google"
          to="/"
        >
          <Link
            className="link button google"
            replace={false}
            target={false}
            title="Sign up with Google"
            to="/"
          >
            <a
              className="link button google"
              href="/"
              onClick={[Function]}
              target={false}
              title="Sign up with Google"
            >
              <span>
                Sign up
                 with 
                Google
              </span>
            </a>
          </Link>
        </Link>
      </div>
    </Flex>
  </Row>
</OauthButton>
`;

exports[`OauthButton should render login correctly 1`] = `
<OauthButton
  className="testClassString"
  displayName="Google"
  login={true}
  service="google"
  to="/"
>
  <Row
    className="testClassString"
    full={true}
  >
    <Flex
      className="testClassString"
      flow="row"
      full={true}
    >
      <div
        className="flex row undefined full testClassString"
      >
        <Link
          className="button google"
          newTab={false}
          target="_self"
          title="Login with Google"
          to="/"
        >
          <Link
            className="link button google"
            replace={false}
            target={false}
            title="Login with Google"
            to="/"
          >
            <a
              className="link button google"
              href="/"
              onClick={[Function]}
              target={false}
              title="Login with Google"
            >
              <span>
                Login
                 with 
                Google
              </span>
            </a>
          </Link>
        </Link>
      </div>
    </Flex>
  </Row>
</OauthButton>
`;
